### PR TITLE
/settings sidebar doesn't match figma designs fix

### PR DESCRIPTION
**Pull request by harshsinghatz**
## What does this PR do?

changes made:
- fix: the spacing in the settings sidebar (according to the figma design given in the issue mentioned below)
- fix: team logo positioning in mobile settings sidebar and also give round shape (according to design)

Fixes # (issue)
[Related Issue](https://github.com/calcom/cal.com/issues/4643) - Related issue

image

image

**Environment**: Staging(main branch) / Production

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] E2E
- [ ] Local testing ( navigate to settings tab and verify for the figma design changes )

### CODE REVIEW

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 1]
```diff
@@ -120,44 +120,46 @@ const SettingsSidebarContainer = ({ className = "" }) => {
...
             textClassNames="text-md font-medium leading-none text-black"
+            className="my-6"
           />
         </div>
         {tabsWithPermissions.map((tab) => {
           return tab.name !== "teams" ? (
             <React.Fragment key={tab.href}>
-              <div>
-                <div className="group flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900">
+              <div className={`${!tab.children?.length ? "!mb-3" : ""}`}>
```
Comment (sean-brydon) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:133]: Overall looks good - great job. However, I can't figure out this line is doing -> I've tried removing it and have noticed no difference.

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 2]
```diff
@@ -120,44 +120,46 @@ const SettingsSidebarContainer = ({ className = "" }) => {
...
                     disableChevron
                   />
                 ))}
               </div>
             </React.Fragment>
           ) : (
             <React.Fragment key={tab.href}>
-              <div>
-                <div className="group mt-2 flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900">
+              <div className={`${!(tab.children ?? []) && "!mb-3"}`}>
```
Comment (sean-brydon) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:157]: Same as above

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 1 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:133]: For example, if there are no children in the tab, we would add the "mb-3" class, but if there are some children, we don't want the extra margin. The same logic applies below

Comment (sean-brydon) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:133]: Okay sweet - is there any reason it is set as important? There is nothing else assigning mb on this div so I don't believe it is needed

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 2 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:157]: Here I think there is a mistake, I should check for the length and not the array. I will fix this!!!

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 1 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:133]: Actually i tried removing it, but it doesn't work that way (not sure about the reason)

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 2 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:157]: I will remove the important here

File: packages/ui/v2/core/layouts/SettingsLayout.tsx [hunk 3]
```diff
@@ -120,44 +120,46 @@ const SettingsSidebarContainer = ({ className = "" }) => {
...
+                    className={`my-0.5 h-7 ${tab.children && index === tab.children?.length - 1 && "!mb-3"}`}
                     disableChevron
                   />
                 ))}
               </div>
             </React.Fragment>
           ) : (
             <React.Fragment key={tab.href}>
-              <div>
-                <div className="group mt-2 flex h-9 w-64 flex-row items-center rounded-md px-3 py-[10px] text-sm font-medium leading-none text-gray-600 hover:bg-gray-100  group-hover:text-gray-700 [&[aria-current='page']]:bg-gray-200 [&[aria-current='page']]:text-gray-900">
```
Comment (sean-brydon) [inline packages/ui/v2/core/layouts/SettingsLayout.tsx:156]: Also I'm pretty sure we need these styles still 

So we can have active/hover states on the items

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Please also remove yarn.lock from this commit - not needed

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 3 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:156]: Every other styles are still present

**Comment by harshsinghatz** [comment]
@sean-brydon Done with changes!, lmk if there are any other feedback changes

File: packages/ui/v2/core/layouts/SettingsLayout.tsx (hunk 3 above)
Comment (harshsinghatz) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:156]: I reverted the yarn.lock with the master version

Comment (sean-brydon) [reply packages/ui/v2/core/layouts/SettingsLayout.tsx:156]: > Every other styles are still present

Must have been a git bug - showed the line as deleted for me :)

**Review by sean-brydon — CHANGES_REQUESTED** [review]
LGTM - nice job man :) - 

Yarn.lock still seems to be present for some reason - looks like a new line

Will approve once this has been fixed

**Comment by harshsinghatz** [comment]
> LGTM - nice job man :) -
> 
> Yarn.lock still seems to be present for some reason - looks like a new line
> 
> Will approve once this has been fixed

Thanks:)
Changes are done yarn.lock is not present now

**Review by sean-brydon — APPROVED** [review]
Awesome :)

**Comment by harshsinghatz** [comment]
Sean, just a doubt. When will this PR be merged I am unfamiliar with the process (in cal.com)

**Comment by sean-brydon** [comment]
> Sean, just a doubt. When will this PR be merged I am unfamiliar with the process (in cal.com)

This will be merged into main once all checks have passed and branch is up to date - then it will be in production in our next release :) Thanks again

**Pull request merged**